File: inst/expt/portfolio.py
import numpy as np
##np.random.seed(123)
n = 10
# mu = np.abs(np.random.randn(n, 1))
# Sigma = np.random.randn(n, n)
# Sigma = Sigma.T.dot(Sigma)

Sigma = np.genfromtxt("Sigma.csv", delimiter=",")
mu = np.genfromtxt("mu.csv", delimiter=",")

# Long only portfolio optimization.
from cvxpy import *
w = Variable(n)
gamma = Parameter(sign='positive')
gamma.value = 1
ret = mu.T*w 
risk = quad_form(w, Sigma)

constraints =  [sum_entries(w) == 1, 
                w >= 0]

prob = Problem(Maximize(ret - gamma*risk),
               constraints)
soln = prob.solve(verbose = True)

constraints[0].dual_value
constraints[1].dual_value

File: inst/expt/portfolio.R
library(cvxr)
n <- 10
set.seed(123)
mu <- abs(rnorm(n))
Sigma <- matrix(rnorm(n*n), nrow=n)
Sigma <- t(Sigma) %*% Sigma

# Long only portfolio optimization.
w <- Variable(n)
ret <- t(mu) %*% w
risk <- quad_form(w, Sigma)

constraints <- list(SumEntries(w) == 1,
                    w >= 0)

prob <- Problem(Maximize(ret - risk),
                constraints)
soln <- solve(prob, verbose = TRUE)

soln$getDualValue(constraints[[1]])
soln$getValue(w)

soln$getDualValue(constraints[[2]])

